### ISSUE TITLE
about loss

### ISSUE THREAD

**Issue by GuobinZhangTJU** [MccreeZhao/QAMFace#8]
the loss function used in the project was loss = self.focalLoss(thetas, labels)??? not Additive Angular Margin Loss?

**Comment by MccreeZhao** [comment]
Hi~ Thanks for your question.
I am not sure if I truly understand your question...But I could try to give a pipeline of face recognition as below: 
Img ->Preprocess-> Backbone -> Feature -> ArcFace/QAMFace -> Probability -> CrossEntropy/Focal

Hope that will be helpful...

**Comment by GuobinZhangTJU** [comment]
thanks for your reply. my question is:
the introduction saied that is a Pytorch implementation for Quadratic Additive Angular Margin Loss for Face Recognition. 
however, the loss used in this project seems to be focal loss, not Additive Angular Margin Loss. I am not sure if I am right.
thanks again.
waiting for your reply.

**Comment by GuobinZhangTJU** [comment]
> Hi~ Thanks for your question.
> I am not sure if I truly understand your question...But I could try to give a pipeline of face recognition as below:
> Img ->Preprocess-> Backbone -> Feature -> ArcFace/QAMFace -> Probability -> CrossEntropy/Focal
> 
> Hope that will be helpful...

thanks for your reply. my question is:
the introduction saied that is a Pytorch implementation for Quadratic Additive Angular Margin Loss for Face Recognition.
however, the loss used in this project seems to be focal loss, not Additive Angular Margin Loss. I am not sure if I am right.
thanks again.
waiting for your reply.

**Comment by MccreeZhao** [comment]
![image](https://user-images.githubusercontent.com/26180075/95553122-79a56800-0a40-11eb-8f8c-40af5afabf19.png)


Actually, AM/QAM loss is a preprocess. And focal loss is connected with cross-entropy. QAMFace and Focal loss are two independent phases.
